mllib/src/main/scala/org/apache/spark/ml/classification/GBTClassifier.scala

```diff
@@ -248,12 +269,38 @@ class GBTClassificationModel private[ml](
     if (prediction > 0.0) 1.0 else 0.0
   }
 
+  override protected def predictRaw(features: Vector): Vector = {
+    val treePredictions = _trees.map(_.rootNode.predictImpl(features).prediction)
+    val prediction = blas.ddot(numTrees, treePredictions, 1, _treeWeights, 1)
```

Comment:
it looks like BLAS.dot is only for Vector, but these are both arrays.  I'm worried that this may degrade performance.  Is this specifically what you are looking for:
BLAS.dot(Vectors.dense(treePredictions), Vectors.dense(_treeWeights))
is the extra dense vector allocation worth it?